python/src/aiconfig/editor/client/src/components/AIConfigEditor.tsx

```diff
@@ -140,9 +140,21 @@ export default function AIConfigEditor({
   const [serverStatus, setServerStatus] = useState<"OK" | "ERROR">("OK");
   const [aiconfigState, dispatch] = useReducer(
     aiconfigReducer,
-    aiConfigToClientConfig(initialAIConfig)
+    aiConfigToClientConfig(providedAIConfig)
   );
 
+  const [prevProvidedConfig, setPrevProvidedConfig] =
+    useState<AIConfig>(providedAIConfig);
+  // After initializing the aiconfigState, we should also support updating the
+  // state if the provided AIConfig changes externally
```

Comment:
Nice, I like this solutino because it doesn't change localEditor implementation either since that only sets aiconfig initial state on load api call